### DISCUSSION TITLE
How does one write unit tests for components using Vue-Apollo

### DISCUSSION THREAD

**Discussion by JasonLi914** [Akryum/vue-apollo#286] [Q&A]
As the title suggests, does anyone have an example somewhere where they successfully mock out vue-apollo for unit tests?

**Comment by gmac** [comment]
Bump on this question. It would be really helpful to have some documented guidance on this. @JasonLi914 – did you figure out any solutions that you could report back here with?

**Comment by rkomiyama** [comment]
Also bumping this question. I'm not sure if there's a way to achieve 100% code coverage.

<img width="333" alt="screen shot 2018-08-20 at 3 03 39 pm" src="https://user-images.githubusercontent.com/347483/44360971-76c71200-a48a-11e8-9190-88695b1d4f1c.png">

**Update**
I figured it out! When mounting a local Vue, also mount an apollo provider. For example...

```
const apolloClient = new ApolloClient({
	link: new HttpLink({
		uri: BASE_API_URL + ':' + BASE_API_URL_PORT + BASE_GQL_PATH,
		fetch
	}),
	cache: new InMemoryCache()
});

const apolloProvider = new VueApollo({
	defaultClient: apolloClient
});

describe('SearchForm.vue', () => {
	beforeEach(() => {
		const localVue = createLocalVue();
		localVue.use(Vuetify);
		localVue.use(VueApollo);

		cmp = shallowMount(SearchForm, {
			provide: apolloProvider.provide(),
			localVue
		});
	});
});
```

**Comment by nathany** [comment]
It appears that `.provide()` is deprecated now
